flake8_trio/visitors/visitor91x.py

```diff
@@ -329,37 +496,33 @@ def visit_loop(self, node: ast.While | ast.For | ast.AsyncFor):
             self.uncheckpointed_statements.update(self.uncheckpointed_before_break)
 
         # reset break & continue in case of nested loops
-        self.set_state(outer)
+        self.outer[node]["uncheckpointed_statements"] = self.uncheckpointed_statements
+        self.restore_state(node)
 
-    visit_While = visit_loop
-    visit_For = visit_loop
-    visit_AsyncFor = visit_loop
+    leave_For_orelse = leave_While_orelse
 
     # save state in case of continue/break at a point not guaranteed to checkpoint
-    def visit_Continue(self, node: ast.Continue):
+    def visit_Continue(self, node: cst.Continue):
         if not self.async_function:
             return
         self.uncheckpointed_before_continue.update(self.uncheckpointed_statements)
 
-    def visit_Break(self, node: ast.Break):
+    def visit_Break(self, node: cst.Break):
+        self.has_break = True
         if not self.async_function:
             return
         self.uncheckpointed_before_break.update(self.uncheckpointed_statements)
 
     # first node in a condition is always evaluated, but may shortcut at any point
     # after that so we track worst-case checkpoint (i.e. after yield)
-    def visit_BoolOp(self, node: ast.BoolOp):
+    def visit_BooleanOperation_right(self, node: cst.BooleanOperation):
```

Comment:
minor logic diff, ast will save `x and y and z` as a single `And` with three values, but cst always parses boolops as binary trees. This leads to slight differences in behaviour as well - see eval files